### DISCUSSION TITLE
Enhancement: Custom search engine

### DISCUSSION THREAD

**Discussion by daes10** [linuxserver/Heimdall#993] [Q&A] [answered]
Is there a way to implement a custom search engine? (A inserted link or something)
Or can anyone add "Brave Search" to the search engines?

https://search.brave.com/

**Comment by Dictynna-PHA** [comment, marked as answer]
You can edit the searchproviders.yaml file to use any provider to pin as a homepage search.

In docker the file is in the www folder.

brave:
   id: brave
   url: https://search.brave.com/search
   name: Brave
   method: get
   target: _blank
   query: q


~~That said I've had an issue where I can only use 5 providers, if I try adding a 6th then settings crash and gives an error 500 on the setting page.
Potentially I mesed up the formatting but I've not had time to double check yet.~~

Edit: Refresh apps cured the above.

Alternatively you can just add it as an 'application' and type in the URL in the requisite box although that only takes you to the page.

**Comment by KodeStar** [comment]
https://github.com/linuxserver/Heimdall/discussions/categories/search-providers for more options
